Shopify/cli
# Add `--force` flag for validation on `shopify theme dev/pull/push/share` if users are running the command in a theme directory

### WHY are these changes introduced?

- Fixes https://github.com/Shopify/shopify-cli/issues/2669
- Addition to https://github.com/Shopify/shopify-cli/pull/2680

We want to…
- ensure that partners execute `shopify theme serve/dev` in the _correct_ directory
- enforce the usage of the theme [architecture](https://shopify.dev/themes/architecture#directory-structure-and-component-types) directory structure

### WHAT is this pull request doing?

Add `--force` flag for validation on `shopify theme dev` if users are running the command in a theme directory.

### How to test your changes?

See https://github.com/Shopify/shopify-cli/pull/2680, but test with `shopify theme dev`, instead of `shopify theme serve`.
Make sure that your CLI 2.x version has the correct changes.

### Post-release steps

- [x] <s>Update shopify.dev docs, as `--force` flag has been added.</s> – flag is hidden, so no need to update docs f.t.m.

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
